lib/semantics/resolve-names.cc

```diff
@@ -2834,14 +2852,14 @@ Symbol *DeclarationVisitor::FindExplicitInterface(const parser::Name &name) {
 const Symbol *DeclarationVisitor::FindTypeSymbol(const parser::Name &name) {
   for (const Scope *scope{&currScope()};;) {
     CHECK(scope->kind() == Scope::Kind::DerivedType);
-    if (const auto *symbol{FindInScope(*scope, name)}) {
+    if (const Symbol * symbol{FindInScope(*scope, name)}) {
       return symbol;
     }
-    const auto *extends{scope->symbol()->get<DerivedTypeDetails>().extends()};
-    if (!extends) {
+    if (const Symbol * parent{scope->symbol()->GetParent()}) {
+      scope = parent->scope();
+    } else {
       return nullptr;
     }
-    scope = extends->scope();
```

Comment:
I think this was clearer with the "increment" expression as the last statement of the body of the loop.

Reply:
OK, will revert.